src/js_sync.hpp

```diff
@@ -310,6 +310,28 @@ class SyncSessionErrorBase {
 public:
     virtual typename T::Function func() = 0;
     virtual void operator()(std::shared_ptr<SyncSession>, SyncError){};
+protected:
+    static typename T::Object create_error_object(typename T::Context ctx, const SyncError& error)
+    {
+        std::string name = "SyncError";
+        auto error_object = Object<T>::create_empty(ctx);
```

Comment:
This object should have the global `Error` set as its prototype, such that `instanceof Error` will succeed as expected:

```
try {
  // failing code
} catch (err) {
  if (err instanceof Error) {
    console.error(err.message);
  }
}